src/utilities/geometry/Polyhedron.cpp

```diff
@@ -268,6 +268,25 @@ void Polyhedron::performEdgeMatching() {
       }
     }
   }
+
+  // #5002 - special case to find edges that are used to "cut" in to a surface to remove an interior hole
+  // we allow these edges to double count the first surface since they bound the same surface on two sides
+  for (auto& surface : m_surfaces) {
+    auto& edges = surface.edges;
+    for (size_t i = 0; i < edges.size(); ++i) {
+      for (size_t j = 0; j < edges.size(); ++j) {
```

Comment:
I know I'm guilty here because I've done it a couple of lines above when looping on surfaces, but we can avoid traversing that much by just doing **combinations** (instead of **permutations**)

```suggestion
    for (size_t i = 0; i < edges.size() - 1; ++i) {
      for (size_t j = i + 1; j < edges.size(); ++j) {
```